### DISCUSSION TITLE
What is the right way of reloading pyodide after fatal error

### DISCUSSION THREAD

**Discussion by roubkar** [pyodide/pyodide#3350] [Q&A]
Hey folks, `pyodide` is an amazing technology.
I've started to use it lately but faced a problem, after running multiple executions it throws a `memory access out of bounds` error and after that `fatal error`.

According to this PR https://github.com/pyodide/pyodide/pull/2391 it should be possible to continue using pyodide after loading it again, however, it still throws errors.

Could you please help me with resolving this issue?

**Comment by hoodmane** [comment]
1. Can you make a report explaining what triggers the fatal errors?
2. Did you try
```js
let new_pyodide = load Pyodide();
new_pyodide.runPython("1+2");
```
If you think you tried something like this and it didn't work please give a code example so we can see if it's a bug or if there's some detail that you need to be doing differently or what.

**Reply by roubkar** [reply]
Hey @hoodmane, thanks for a fast reply!

I'm doing smth like this:
```javascript
let pyodide = await loadPyodide();
pyodide.runPython("1+2");
...
pyodide._api.fatal_error = async () => {
    pyodide = await loadPyodide();
}
```

will try to share an example for reproducing.

**Reply by hoodmane** [reply]
Hmmm I have to admit that looks a bit scary to me. Have you thought about avoiding fatal errors instead of dealing with them when they happen?

`from _pyodide_core import trigger_fatal_error` may be useful.

**Reply by roubkar** [reply]
I'm building an API for describing layouts with data visualization components, where function casting happens, so I'm getting this error after some time:
<img width="1607" alt="Screen Shot 2022-12-13 at 21 00 02" src="https://user-images.githubusercontent.com/9951653/207399735-5525c302-35a7-41b1-87a5-6ca88cbf8de6.png">

**Comment by gabrielfougeron** [comment]
You could run Pyodide in a separate web worker. This would allow you to simply kill the worker and setup another one.

**Reply by roubkar** [reply]
yeah, I was thinking of that, seems for interrupting the execution I also need to use pyodide in a worker.

But here comes another question: can I use a single namespace to pass as globals during `runPython` in a worker?
